index.js

```diff
@@ -1,20 +1,259 @@
+let playerObject = {
+  //initialize player inventory
+  playerInventory: [],
+  //global room location
+  currentRoom: "Room #1",
+};
+
+//a class for items to pick up in the game
+class Item {
+  constructor(name, description, action, takeable) {
+    this.name = name;
+    this.description = description;
+    this.action = action || "nothing"; //if item does not have an action property nothing happens
+    this.takeable = takeable || "false"; //if item is not takeable returns false
+  }
+
+  //a take function method on class Item to take player from one room to the next
+  take() {
+    if (this.takeable === "true") {
+      playerInventory.push(this.name); //this pushes to inventory which is initialized as a global variable
+      console.log(`You picked up ${this.name} and it is in your inventory.`);
+    } else {
+      console.log(`You can't do that action`);
+    }
+  }
+
+  //a use function method on class Item to use items in the game
+  use() {
+    if (this.name === "key" && roomInventory.includes("key")) {
+      console.log("You can open the door to the Puzzle Room.");
+    } else {
+      console.log(this.action);
+    }
+  }
+  //an examine function method on class Item to examine items in the game
+  examine() {
+    console.log(`you see ${this.description}`);
+  }
+  //a drop function method on class Item to drop items in the game
+  drop() {
+    if (this.takeable === "true" && playerInventory.includes(this.name)) {
+      inventoryPlayer.pop();
+      return `You dropped ${this.name}`;
+    } else {
+      return `You can't do that action`;
+    }
+  }
+}
+
+//initialize items to use in the game
+//creates a door item
+let door = new Item(
+  "door", //name
+  "a large, austere door looming in front of you that appears to need a key", //description
+  "the door is unlocked", //action
+  false //takeable
+);
+
+//creates a key item
+let key = new Item(
+  "key", //name
+  "something glinting, there is a key in the hollow of a tree", //description
+  "you turn the key", //action
+  true //takeable
+);
+
+//creates a mushroom item
+let mushroom = new Item(
+  "mushroom", //name
+  "a glowing mushroom", //description
+  "you take the mushroom", //actions
+  true //takeable
+);
+
+//creates a sign item
+let sign = new Item(
+  "sign", //name
+  "a sign posted on a tree", //description
+  "you take the sign", //actions
+  true //takeable
+);
+
+//item lookup table (looks up string version and spits out variable version allowing the variable version to be looked up)
+let lookupItemTable = {
+  door: door,
+  key: key,
+  mushroom: mushroom,
+  sign: sign,
+};
+
+//room class to create rooms for the game
+class Room {
+  constructor(roomName, roomDescription, locked, item, connections) {
+    this.roomName = roomName;
+    this.roomDescription = roomDescription;
+    this.locked = locked;
+    this.item = item;
+    this.connections = connections;
+  }
+
+  //move method on class Room to move between rooms
+  move() {
```

Comment:
I think this works but can't tell without the message telling me where I'm at and the description... but it does print the array of connecting rooms for certain rooms so I think it's working...